sound/soc/sof/intel/hda.c

```diff
@@ -643,12 +641,8 @@ static irqreturn_t hda_dsp_interrupt_handler(int irq, void *context)
 {
 	struct snd_sof_dev *sdev = context;
 
-	/* clear flags for interrupt sources */
-	sdev->irq_event = 0;
-
-	if (hda_dsp_check_stream_irq(sdev) ||
-	    hda_dsp_check_ipc_irq(sdev)    ||
-	    hda_dsp_check_sdw_irq(sdev)) {
+	if (snd_sof_dsp_read(sdev, HDA_DSP_HDA_BAR, SOF_HDA_INTSTS) &&
+	    SOF_HDA_INT_GLOBAL_EN) {
```

Comment:
@bardliao is this correct? Whats the point of && SOF_HDA_INT_GLOBAL_EN? This will always be true isnt it?

Reply:
> @bardliao is this correct? Whats the point of && SOF_HDA_INT_GLOBAL_EN? This will always be true isnt it?

Good catch @ranj063 fix it now

Reply:
@bardliao I guess the intention is correct here and even the logic but I will not use SOF_HDA_INT_GLOBAL_EN, you should define SOF_HDA_INTSTS_GIS as BIT(31) and use that instead of SOF_HDA_INT_GLOBAL_EN. Otherwise, you're sending the wrong signal

Reply:
yes,  It is a bug that using A() || B() || C() since if A() return true , B() C() will not be called.
I was blocked by this bug when I add D()

Reply:
Hmmm, the logic is correct  here. Ack @ranj063 it's a bit misleading to use SOF_HDA_INT_GLOBAL_EN as the macro name, but SOF_HDA_INT_GLOBAL_STS (when added) will be same BIT(31), so logic is still correct.

Reply:
here, it is better to add some comments.

Reply:
I update my finding.  I will add wake interrupt for HDA & SDW in this function.  I found wake interrupt is not included by SOF_HDA_INT_GLOBAL_STS.  So we need to update a new if for this @kv2019i ?

Reply:
Thanks @ranj063 @kv2019i @RanderWang . The PR has updated.